Add new model for annotation_slim

marcospri opened the pull request:
Migration over in: https://github.com/hypothesis/h/pull/8213

seanh commented on h/models/annotation_slim.py:
```diff
@@ -0,0 +1,70 @@
+from h.db import Base, types
+
+
+class AnnotationSlim(Base):
+    """
+    AnnotationSlim represents fundamental properties of annotations.
```
I think `AnnotationMetadata` is probably the best name.

I can see what you're getting at with "fundamental properties" but I think I'd consider some of the properties that _aren't_ in this table to be fundamental properties of an annotation (e.g. the actual annotation text itself) whereas I think I'd consider some of the properties that _are_ in this table to be _not_ fundamental, e.g. `moderated` etc.

I think "metadata" is probably the word that most closely captures what's in this table.

seanh reviewed (approved):
Some ideas that we might want to do now or in the future:

* We might want to add some more columns into the foreign key constraint to `annotation` just to enforce that those columns can't differ between the two tables: `created`, `updated`, `deleted`, `shared`, `moderated`, `document_id`
* We might want to include copies of some of the metadata columns from the `user` and `group` tables directly in this table in the same way as we do with some columns from the `annotation` table. I think we'd only do this if it helped to make some queries that the app actually does more efficient by avoiding joins to `user` and `group`. It does seem like a weird thing to do so we may want to avoid it unless necessary
* We might want to add some more indexes to this table than what you have here. Again, depends on what queries we actually end up doing and whether there are any more indexes that would actually help. Also reminding myself that I think the foreign key constraints also create indexes on those columns

marcospri commented:
> We might want to add some more columns into the foreign key constraint to annotation just to enforce that those columns can't differ between the two tables: created, updated, deleted, shared, moderated, document_id

We will have to create an index on the original table on this. I might be to expensive for too little benefit.


> We might want to include copies of some of the metadata columns from the user and group tables directly in this table in the same way as we do with some columns from the annotation table. I think we'd only do this if it helped to make some queries that the app actually does more efficient by avoiding joins to user and group. It does seem like a weird thing to do so we may want to avoid it unless necessary

I'd only do that if it absolutely necessary. JOINs are fast at our scale.


> We might want to add some more indexes to this table than what you have here. Again, depends on what queries we actually end up doing and whether there are any more indexes that would actually help. Also reminding myself that I think the foreign key constraints also create indexes on those columns

Yeah pretty much all column (all?) have index. We could create few compound ones if we need to use them.
